src/engraving/libmscore/edit.cpp

```diff
@@ -3478,6 +3490,9 @@ void Score::cmdDeleteSelection()
             select(cr, SelectType::SINGLE);
         }
     }
+    if (keySig) {
+        select(keySig, SelectType::SINGLE);
+    }
```

Comment:
I must admit I'm not entirely sure if the key signature selection should override all other selections. :thinking:

I didn't find it a problem in the tests I did, but let me know if you'd like this fine-tuned further.